File: Arrays/LongestConsecSeq/LongestConsecSeq.py
class Solution(object):
    def longestConsecutive(self, nums):
        nums_set, max_seq_length, curr_seq_length = set(nums), 0, 0
        for num in nums:
            curr_seq_length = 0
            if num - 1 in nums_set:
                continue
            curr_seq_length += 1
            curr_num = num + 1
            while curr_num in nums_set:
                curr_seq_length += 1
                curr_num += 1
            max_seq_length = max(max_seq_length, curr_seq_length)
        return max(max_seq_length, curr_seq_length)

File: Arrays/LongestConsecSeq/LongestConsecSeq.go
func longestConsecutive(nums []int) int {
	maxSeq, currSeq := 0, 0
	set := arrayToSet(nums)
	for _, num := range nums {
		currSeq = 0
		if set[num-1] {
			continue
		}
		currSeq += 1
		currNum := num + 1
		for set[currNum] {
			currSeq += 1
			currNum += 1
		}
		if currSeq > maxSeq {
			maxSeq = currSeq
		}
	}
	if currSeq > maxSeq {
		maxSeq = currSeq
	}
	return maxSeq
}

func arrayToSet(nums []int) map[int]bool {
	set := map[int]bool{}
	for _, num := range nums {
		if !set[num] {
			set[num] = true
		}
	}
	return set
}